Refactor store: add type defs for zustand stores

vedantnn71 opened the pull request:
I have:
- [x] Added type definitions for the stores and removed any/s

subhoghoshX commented:
Thanks for working on it vedant. Unfortunately, I might be able to merge it. As I'm working on refactoring the store and the methods for a while now.

You can slim down PR to only include types i.e. everywhere I used `any`, replace that with the proper types.

vedantnn71 commented:
Hey @subhoghoshX, I've reverted back to previous commit and moved all stores in just one file as they were previously, now only types are added and a lot of any/s are removed in this pr.

subhoghoshX commented:
There are too many changes to look at. Please, revert back the store file to its previous state. Only change the types.

vedantnn71 commented:
I have just:
1. Replaced explicit return statements w/ implicit returns
2. Renamed capitalized x and y to lowercase 
3. And removed any/s used while using stores in the components

Do you want to revert lowercase x and y back to capitalized ones?

subhoghoshX commented:
You might have moved things up and down. The diff looks confusing.

vedantnn71 commented:
> You might have moved things up and down. The diff looks confusing.

now?

subhoghoshX commented:
Try to sync your branch pls.

subhoghoshX commented on components/TweetCard/index.tsx:
```diff
@@ -32,7 +32,7 @@ export default function TweetCard({ rootRef }: any) {
   function mouseMoveHandler(e: any) {
     if (mouseDown && spaceDown) {
-      setMoveBy({ X: moveBy.X + e.movementX, Y: moveBy.Y + e.movementY });
+      setMoveBy({ x: moveBy.x + e.movementX, y: moveBy.y + e.movementY });
```
```suggestion
      setMoveBy({ x: moveBy.X + e.movementX, y: moveBy.Y + e.movementY });
```

subhoghoshX commented on components/TweetCard/index.tsx:
```diff
@@ -42,7 +42,7 @@ export default function TweetCard({ rootRef }: any) {
       onMouseMove={mouseMoveHandler}
       style={{
-        transform: `translate(${moveBy.X}px, ${moveBy.Y}px)`,
+        transform: `translate(${moveBy.x}px, ${moveBy.y}px)`,
```
```suggestion
        transform: `translate(${moveBy.X}px, ${moveBy.Y}px)`,
```

subhoghoshX commented on store/index.ts:
```diff
@@ -1,143 +1,135 @@
-      set((state: any) => ({ width: c }));
+export const useCardStore = create<CardState>((set) => ({
+  height: 332,
+  width: 672,
```
bring back `rounded` because it doesn't work.

subhoghoshX replied:
bring back `rounded` because it doesn't work anymore.

vedantnn71 commented:
I've made the changes that you said :)

Can I add type defs for tweet store too?

subhoghoshX commented on components/TweetCard/index.tsx:
```diff
@@ -32,7 +32,7 @@ export default function TweetCard({ rootRef }: any) {
   function mouseMoveHandler(e: any) {
     if (mouseDown && spaceDown) {
-      setMoveBy({ X: moveBy.X + e.movementX, Y: moveBy.Y + e.movementY });
+      setMoveBy({ x: moveBy.X + e.movementX, y: moveBy.Y + e.movementY });
```
```suggestion
      setMoveBy({ X: moveBy.X + e.movementX, Y: moveBy.Y + e.movementY });
```

subhoghoshX commented on store/index.ts:
```diff
@@ -1,143 +1,141 @@
+export const usePanStore = create<PanState>((set) => ({
+  spaceDown: false,
+  mouseDown: false,
+  moveBy: { x: 0, y: 0 },
```
```suggestion
  moveBy: { X: 0, Y: 0 },
```

subhoghoshX commented on store/types/panState.ts:
```diff
@@ -0,0 +1,13 @@
+}
+
+export type MoveBy = {
+  x: number;
+  y: number;
```
```suggestion
  X: number;
  Y: number;
```

subhoghoshX commented:
> I've made the changes that you said :)
> 
> Can I add type defs for tweet store too?

Yeah sure :)

vedantnn71 commented:
Done!

I think now the pr is ready to be merged

subhoghoshX commented:
Thank you @vedantnn71.

subhoghoshX commented:
Thanks, dude.
